test/unit/person_test.rb

```diff
@@ -1407,4 +1408,327 @@ def test_updated_not_changed_when_adding_notifiee_info
+    sop_both = FactoryBot.create :sop
+    sop_other = FactoryBot.create :sop
+    person_to_keep = FactoryBot.create(:person)
+    other_person = FactoryBot.create(:max_person)
+    FactoryBot.create :subscription, person: person_to_keep, subscribable: sop_keep
+    FactoryBot.create :subscription, person: person_to_keep, subscribable: sop_both
+    FactoryBot.create :subscription, person: other_person, subscribable: sop_both
+    FactoryBot.create :subscription, person: other_person, subscribable: sop_other
+    orig_subs = person_to_keep.subscriptions.pluck(:subscribable_id)
+    other_subs = other_person.subscriptions.pluck(:subscribable_id)
+    assert orig_subs.include?(sop_keep.id)
+    assert orig_subs.include?(sop_both.id) && other_subs.include?(sop_both.id)
+    assert other_subs.include?(sop_other.id)
+
+    disable_authorization_checks { person_to_keep.merge(other_person) }
+    person_to_keep.reload
+
+    assert_equal (orig_subs + other_subs).compact.uniq.sort, person_to_keep.subscriptions.pluck(:subscribable_id).sort
+  end
+
+  test 'merge resources without duplication' do
+    person_to_keep = FactoryBot.create(:person)
+    other_person = FactoryBot.create(:person)
+    s1 = FactoryBot.create(:sop, contributor: person_to_keep)
+    s2 = FactoryBot.create(:sop, contributor: person_to_keep, creator_ids: [person_to_keep.id, other_person.id])
+    s3 = FactoryBot.create(:sop, contributor: other_person)
+    s4 = FactoryBot.create(:sop, contributor: other_person, creator_ids: [person_to_keep.id, other_person.id])
+    s5 = FactoryBot.create(:sop, contributor: FactoryBot.create(:person), creator_ids: [other_person.id])
+
+    disable_authorization_checks { person_to_keep.merge(other_person) }
+    person_to_keep.reload
+
+    assert_equal [s1, s2, s3, s4].sort, person_to_keep.contributed_sops.sort
+    assert_equal [s2, s4, s5].sort, person_to_keep.created_sops.sort
+  end
+
+  test 'merge all types of resources' do
+    person_to_keep = FactoryBot.create(:person)
+    other_person = FactoryBot.create(:person)
+
+    types_without_creators = %w[Event Strain]
+    Person::RELATED_RESOURCE_TYPES.each do |resource_type|
+      sym = resource_type == 'SampleType' ? :simple_sample_type : resource_type.underscore.to_sym
+      FactoryBot.create(sym, contributor: other_person)
+      unless types_without_creators.include?(resource_type)
+        FactoryBot.create(sym, contributor: FactoryBot.create(:person), creator_ids: [other_person.id])
+      end
+    end
+
+    disable_authorization_checks { person_to_keep.merge(other_person) }
+    person_to_keep.reload
+
+    resource_types = Person::RELATED_RESOURCE_TYPES.length
+    added_isa_structure = 3 # An additional inv is created for the study, and inv and study are created for the assay
+    assert_equal resource_types + added_isa_structure, person_to_keep.contributed_items.length
+    assert_equal resource_types - types_without_creators.length, person_to_keep.created_items.length
```

Comment:
I don't think it's safe to just check the overall length - one type could be missing and another have a duplicate.
it needs assertions for each type or the contents of the arrays

Reply:
Resolved in [8938b7a](https://github.com/seek4science/seek/pull/1683/commits/8938b7a356e05f286136d8e987a86711d091727c).